frontend/dispatcher.lua

```diff
@@ -12,37 +13,132 @@ local Dispatcher = {
+    zoom = { category="string", event="SetZoomMode", title=_("Zoom to"), args={"contentwidth", "contentheight", "pagewidth", "pageheight", "column", "content", "page"}, toggle={"content width", "content height", "page width", "page height", "column", "content", "page"}, paging=true,},
+
+    --parsed from CreOptions
+    rotation_mode = {category="string", device=true},
+    visible_pages = {category="string", rolling=true},
+    h_page_margins = {category="string", rolling=true},
+    sync_t_b_page_margins = {category="string", rolling=true},
+    t_page_margin = {category="absolutenumber", rolling=true},
+    b_page_margin = {category="absolutenumber", rolling=true},
+    view_mode = {category="string", rolling=true},
+    block_rendering_mode = {category="string", rolling=true},
+    render_dpi = {category="string", rolling=true},
+    line_spacing = {category="absolutenumber", rolling=true},
+    font_size = {category="absolutenumber", title="Font Size", rolling=true},
+    font_weight = {category="string", rolling=true},
+    --font_gamma = {category="string", rolling=true},
+    font_hinting = {category="string", rolling=true},
+    font_kerning = {category="string", rolling=true},
+    status_line = {category="string", rolling=true},
+    embedded_css = {category="string", rolling=true},
+    embedded_fonts = {category="string", rolling=true},
+    smooth_scaling = {category="string", rolling=true},
+    nightmode_images = {category="string", rolling=true},
+}
+
+-- array for item order in menu
+local dispatcher_menu_order = {
+    "show_frontlight_dialog", "toggle_frontlight", "toggle_gsensor", "wifi_on", "wifi_off", "toggle_wifi", "reading_progress", "stats_calendar_view", "history", "open_previous_document", "filemanager", "dictionary_lookup", "wikipedia_lookup", "fulltext_search", "file_search", "full_refresh", "night_mode", "suspend", "exit", "restart", "reboot", "poweroff", "show_menu", "toggle_hold_corners", "toggle_rotation", "wallabag_download", "calibre_search", "calibre_browse_tags", "calibre_browse_series", "favorites", "folder_up", "show_plus_menu", "folder_shortcuts", "prev_chapter", "next_chapter", "first_page", "last_page", "prev_bookmark", "next_bookmark", "go_to", "skim", "back", "previous_location", "latest_bookmark", "clear_location_history", "toc", "bookmarks", "book_statistics", "book_status", "book_info", "book_description", "book_cover", "show_config_menu", "toggle_bookmark", "toggle_inverse_reading_order", "cycle_highlight_action", "cycle_highlight_style", "kosync_push_progress", "kosync_pull_progress", "toggle_page_flipping", "toggle_reflow", "zoom", "rotation_mode", "visible_pages", "h_page_margins", "sync_t_b_page_margins", "t_page_margin", "b_page_margin", "view_mode", "block_rendering_mode", "render_dpi", "line_spacing", "font_size", "font_weight", "font_hinting", "font_kerning", "status_line", "embedded_css", "embedded_fonts", "smooth_scaling", "nightmode_images",
```

Comment:
This on a single line will be unmanageable :)

> I went with using another array for the order.

Might not be the best, we'd have to not forget to update 2 places when adding a new one.

I think defining them once in an array and building your k/v table from it for quick lookup is cleaner (suggested at the bottom of https://github.com/koreader/koreader/pull/6364#issuecomment-657282715).

Reply:
The concern raised in https://github.com/koreader/koreader/pull/6364#issuecomment-657283193 was to avoid duplicating the full table, and instead only duplicating the key ;).

I'm of the mind that the performance point is moot either way, so I kinda like the current approach, but I definitely agree it should be broken down line by line (for readability, and diff-friendliness in case of updates).

Reply:
> the "new" key never makes it to the table, while forgetting to update the actual data table will crash by trying to index a nil or something, right?

Possibly for the crash (haven't checked how it is used), and yes, we won't see it in the menu - but will we always check/have to check if it's present in the menu? :)

But just adding an `id=` to the elements in a single table and using:
```lua
local settingsListById = {}
for _, t in ipairs(settingsList) do
   if t ~= true then
       settingsListById[t.id] = t
   end
end
```
saves 100 lines of code :) (yep, we have nearly that many settings).
(Well, that's just my preference - I like things tidy :)

Reply:
I'm also not one to be bothered by extra LOCs, so, eh ;D.

Reply:
> This on a single line will be unmanageable :)

this was jsut for the first draft untill i got a proper order, was not planning on leaving it this way.

> I think defining them once in an array and building your k/v table from it for quick lookup is cleaner (suggested at the bottom of [#6364 (comment)](https://github.com/koreader/koreader/pull/6364#issuecomment-657282715)).
 
> As for your point about needing to update two things: forgetting to update the order array will essentially mean the "new" key never makes it to the table, while forgetting to update the actual data table will crash by trying to index a nil or something, right?

It will crash.

I like having the k/v table in the order it is in, as it makes adding new things easier, beacuse the items are sorted by where they come from & where they apply as is commented in the table. therefore i would rater have the display order in a seperate array even if it adds loc. (It still will be shorter than ReaderGesture :)

Reply:
> beacuse the items are sorted by where they come from & where they apply as is commented in the table. therefore i would rater have the display order in a seperate array 

OK, that's a convincing enough reason :)
(But may be _where they come from & where they apply_  would be just fine for the menu order ? :)

Reply:
> (But may be _where they come from & where they apply_ would be just fine for the menu order ? :)

not enough, for example: rotation is (now) a device setting and should be listed to the user with the other device settings. however the values are generated by parsing `creopt` so in the k/v it is listed in it's place in `creopt`, this allows a developer making a change in `creopt` to easily see what is affected etc.

Reply:
Then simply the same order as the original menus themselves, in a dumb top-to-bottom, left-to-right order?

Reply:
> Then simply the same order as the original menus themselves, in a dumb top-to-bottom, left-to-right order?

many of them are not in the menus but spread out over the ui (GoTo dialog or frontlight toggle)